src/builder.ts

```diff
@@ -100,10 +100,14 @@ export default class Builder {
 
 			// Catch any errors the stream produces
 			outputStream.on('error', (err: Error) => {
-				self.callHook(hooks, 'buildFailure', handler, err)
+				let layers = self.layers
+				layers.pop()
```

Comment:
The times that they are called are mutually exclusive, but I take your point, probably would be best to copy.